test/node/regress-gh-411.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const test = require('tap');
+
+const jstp = require('../..');
+
+const APP_NAME = 'APP_NAME';
+
+const TOKEN = 'TOKEN';
+
+const interfaces = {
+  iface: {
+    first(connection, token, callback) {
+      connection.session.set(TOKEN, token);
+      callback(null);
+    },
+
+    second(connection, callback) {
+      callback(null, connection.session.get(TOKEN));
+    },
+  },
+};
+
+const application = new jstp.Application(APP_NAME, interfaces);
+const serverConfig = { applications: [application] };
+
+let server;
+let connection;
+
+const reconnector = (connection, reconnectFn) => {
+  if (!connection.closedIntentionally) {
+    reconnectFn('net');
+  }
+};
+
+test.beforeEach(done => {
+  server = jstp.net.createServer(serverConfig);
+  server.listen(0, () => {
+    const port = server.address().port;
+    jstp.net.connect(
+      APP_NAME,
+      { reconnector },
```

Comment:
I'd prefer to have an explicit mention that this is what is actually the core of the test. Because based on `test.test` it looks like we are testing that reconnect is called after the transport destroy. Maybe it's also worth adding a short description at the top.